src/main/java/sdu/se9/tv2/management/system/persistence/PersistenceCredit.java

```diff
@@ -21,46 +23,49 @@ public static PersistenceCredit getInstance() {
         return instance;
     }
 
-    private Persistence persistence = new Persistence("credit.json");
-
-    private int lastID = -1;
-
-    private ArrayList<Credit> credits = new ArrayList<Credit>();
-
-    private PersistenceCredit() {
-        this.read();
-    }
+    private PersistenceCredit() {}
 
     @Override
-    public Credit createCredit(int programID, int personID, String roleName) throws DuplicateRoleNameException {
-        if (this.getCredit(programID, roleName) != null) {
-            throw new DuplicateRoleNameException("Role already exists");
-        }
+    public Credit createCredit(int programID, int personID, String roleName) throws DuplicateRoleNameException, SQLException {
 
-        lastID++;
-        Credit newCredit = new Credit(lastID, programID, personID, roleName);
-        credits.add(newCredit);
+        Connection connection = PersistenceDatabaseHelper.getConnection();
 
-        // Save changes to file
+            if (this.getCredit(programID, roleName) != null) {
+                throw new DuplicateRoleNameException("Role already exists");
+            }
 
-        this.write();
+            PreparedStatement stmt = connection.prepareStatement("INSERT INTO Credit(programID, personID, roleName) VALUES(?,?,?) ", Statement.RETURN_GENERATED_KEYS);
+            stmt.setInt(1,  programID);
+            stmt.setInt(2,  personID);
+            stmt.setString(3,  roleName);
+            stmt.execute();
+            ResultSet rs = stmt.getGeneratedKeys();
+            int i = 0;
+            if (rs.next()) {
+                i = rs.getInt(1);
+            }
 
-        return newCredit;
+            System.out.println("You did it!");
+            System.out.println(i);
+            return new Credit(i, programID, personID, roleName);
+        // Save changes to file
     }
 
     @Override
-    public ArrayList<Credit> getCredits(int programID) {
-        ArrayList<Credit> result = new ArrayList<Credit>();
+    public ArrayList<Credit> getCredits(int programID) throws SQLException {
+        Connection connection = PersistenceDatabaseHelper.getConnection();
 
-        for (int i = 0; i < this.credits.size(); i++) {
-            Credit element = this.credits.get(i);
 
-            if (element.getProgramID() == programID) {
-                result.add(element);
+            PreparedStatement stmt = connection.prepareStatement("SELECT * FROM Credit WHERE programID =" + programID);
```

Comment:
Dette er præcist hvad man aldrig må gøre. I skal bruge prepareStatement ordenligt, og det gøres ved at skrive `programID = ?` i stedet for at tilføje det direkte til strengen, og så bruge `stmt.setInt(1, programID)` for at sætte værdien. Hvis i ikke gør det, så kan i komme til at åbne databasen op for SQL injections.